includes/plugins/class-gravityforms.php

```diff
@@ -0,0 +1,466 @@
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Main class.
+ */
+class GravityForms {
+	/**
+	 * AJAX call result payload.
+	 *
+	 * @var array
+	 */
+	private static $ajax_result_payload = [
+		'status'       => 'success',
+		'confirmation' => '',
+		'errors'       => [],
+	];
+
+	/**
+	 * Classnames used by GF polls plugin are used.
+	 *
+	 * @var string[]
+	 */
+	private static $gf_polls_classnames = [
+		'gpoll_value_selected',
+		'gpoll_button',
+		'gpoll_bar',
+		'gpoll_bar_juice',
+		'gpoll_bar_count',
+		'gpoll_wrapper',
+		'gpoll_ratio_box',
+		'gpoll_ratio_label',
+		'gpoll_field_label',
+		'gpoll_choice_label',
+		'gpoll_container',
+		'gpoll_field',
+		'gpoll_field_number',
+		'gfield',
+		'blue',
+		'green',
+		'red',
+		'orange',
+	];
+
+	/**
+	 * Initialize hooks and filters.
+	 */
+	public static function init() {
+		add_filter( 'newspack_amp_plus_sanitized', [ __CLASS__, 'filter_scripts_for_amp_plus' ], 10, 2 );
+
+		add_filter( 'the_content', [ __CLASS__, 'handle_gf_polls_blocks' ] );
+
+		// The following code makes GravityForms work nicely with AMP.
+		// It's not enough for forms which have conditional logic – these need JS. For this,
+		// AMP Plus is used to allow GF's JS to be sent to the client.
+		//
+		// Adapted from: https://gist.github.com/swissspidy/86e3a50ec5c0f7d46ec4de43824e23a0 .
+		// By Pascal Birchler (https://pascalbirchler.com).
+		add_filter( 'gform_form_args', [ __CLASS__, 'gform_form_args' ] );
+		add_filter( 'gform_field_content', [ __CLASS__, 'gform_field_content' ], 10, 2 );
+		add_filter( 'gform_submit_button', [ __CLASS__, 'gform_submit_button' ], 10, 2 );
+		add_filter( 'gform_pre_render', [ __CLASS__, 'gform_pre_render' ], 10, 2 );
+		add_filter( 'gform_footer_init_scripts_filter', [ __CLASS__, 'gform_footer_init_scripts_filter' ] );
+		add_filter( 'widget_display_callback', [ __CLASS__, 'widget_display_callback' ], 10, 3 );
+		add_filter( 'gform_get_form_filter', [ __CLASS__, 'gform_get_form_filter' ], 10, 2 );
+		add_action( 'gform_enqueue_scripts', [ __CLASS__, 'gform_enqueue_scripts' ], 100 );
+		add_action( 'gform_form_tag', [ __CLASS__, 'gform_form_tag' ], 100 );
+		add_action( 'wp_ajax_gravity_form_submission', [ __CLASS__, 'send_ajax_result' ] );
+		add_action( 'wp_ajax_nopriv_gravity_form_submission', [ __CLASS__, 'send_ajax_result' ] );
+		add_filter( 'gform_confirmation', [ __CLASS__, 'gform_confirmation' ], 10, 3 );
+		add_action( 'gform_post_process', [ __CLASS__, 'gform_post_process' ] );
+		add_filter( 'gform_validation', [ __CLASS__, 'gform_validation' ] );
+		add_filter( 'gform_suppress_confirmation_redirect', [ __CLASS__, 'gform_suppress_confirmation_redirect' ] );
+	}
+
+	/**
+	 * Filter arguments.
+	 *
+	 * @param array $args Arguments.
+	 */
+	public static function gform_form_args( array $args ) {
+		if ( ! is_amp_endpoint() ) {
+			return $args;
+		}
+		$args['ajax']           = false;
+		$args['enableHoneypot'] = '0';
```

Comment:
Why do we need to disable the honeypot field? My understanding is that this is a pretty widely used feature to prevent spam responses.

Reply:
Yeah, I've reverted this change (aff2645c25023ab0b1dfa3c547bf9b4a0833ef24) after testing with a honeypot-enabled form

Reply:
Confirmed that a poll and a form with honeypot fields still captured entries correctly.